### PR TITLE
[DISCOVERY-94] Fix/scan manager unhandled

### PR DESCRIPTION
**Pull request by bruno-fs**

### CODE REVIEW

File: quipucords/conftest.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# pylint: disable=import-outside-toplevel
+from contextlib import suppress
+from unittest.mock import patch
+
+import pytest
+
+
+def _kill_scan_manager(manager_instance):
+    for job in manager_instance.scan_queue:
+        with suppress(AttributeError):
```
Comment (nicolearagao) [inline quipucords/conftest.py:25]: Is there a difference between using contextlib.suppress and a try/catch besides readability?

File: quipucords/conftest.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    from scanner import manager
+
+    scan_manager_instance = manager.Manager()
+    with (
+        patch.object(manager, "sleep"),
+        patch.object(manager, "SCAN_MANAGER", scan_manager_instance),
+    ):
+        yield scan_manager_instance
+
+    _kill_scan_manager(scan_manager_instance)
```
Comment (nicolearagao) [inline quipucords/conftest.py:53]: so after the yield we have the __exit__ and you called the func _kill_scan_manager to do it. Right?

File: quipucords/conftest.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+
+@pytest.fixture(autouse=True)
+def _patch_scan_manager(scan_manager):
+    with (
+        patch.object(scan_manager.__class__, "__call__", _return_itself),
+        patch("api.signal.scanjob_signal.manager.Manager", scan_manager),
+        patch("api.signal.scanjob_signal.manager.SCAN_MANAGER", scan_manager),
+    ):
+        yield
```
Comment (nicolearagao) [inline quipucords/conftest.py:63]: What happens after the with block finishes on this one?

File: quipucords/conftest.py (hunk 1 above)
Comment (bruno-fs) [reply quipucords/conftest.py:25]: nope. this is exactly the same as 
```
try:
   job.kill()
except AttributeError:
    pass
```
just less verbose

File: quipucords/conftest.py (hunk 2 above)
Comment (bruno-fs) [reply quipucords/conftest.py:53]: in a pytest fixture, think about everything before yield as a setup (`TestCase.setUp` equivalent) and what comes after as a teardown (`TestCase.tearDown`)

File: quipucords/conftest.py (hunk 3 above)
Comment (bruno-fs) [reply quipucords/conftest.py:63]: you mean the context with the patches? if so, everything goes back to they original behaviour before the patching

Comment (nicolearagao) [reply quipucords/conftest.py:63]: ok!

**Review by nicolearagao — APPROVED** [review]
LGTM =)

**Review by gtanzillo — APPROVED** [review]

**Pull request merged**